meta-resin-common/recipes-support/resin-logs/resin-logs/resin-logs

```diff
@@ -4,12 +4,26 @@ set -e
 
 . /usr/sbin/resin-vars
 
-if [ "$PERSISTENT_LOGGING" == "true" ]; then
-    echo "[INFO] Configure persistent logs in /mnt/conf/root-overlay/var/log/journal"
-    echo -e "[Journal]\nStorage=persistent\n" > /etc/systemd/journald.conf.d/resin-logs.conf
-elif [ "$PERSISTENT_LOGGING" == "false" ]; then
-    echo "[INFO] Configure volatile logs in /run/log/journal"
-    echo -e "[Journal]\nStorage=volatile\n" > /etc/systemd/journald.conf.d/resin-logs.conf
+machineID=`cat /etc/machine-id`
+
+if [ "$PERSISTENT_LOGGING" == "true" ] && [ ! -d /var/log/journal ]; then
+    systemctl start var-log-journal.mount
+    # Send flush signal to journald
+    killall -USR1 systemd-journald
+elif [ -d /var/log/journal ]; then
+    systemctl stop systemd-journald-dev-log.socket
+    systemctl stop systemd-journald.socket
+    systemctl stop systemd-journald.service
+    systemctl stop var-log-journal.mount
+
+    # Copy persistent logs to ram to keep them consistent
+    mkdir -p /run/log/journal && mv /var/log/journal/$machineID /run/log/journal
```

Comment:
Please have a look at the standard: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html
The -e setting shall be ignored when executing the compound list following the while, until, if, or elif reserved word, a pipeline beginning with the ! reserved word, or any command of an AND-OR list other than the last.